### PR TITLE
Issues #1247 and #807

### LINKED ISSUE #1247: TreeNode.prune does not collapse single descendants from root properly
I noticed this while working on a collapse [method](https://github.com/wasade/yummy-octo-duck/blob/master/ipynb/Tree%20array%20shearing.ipynb) (i.e., prune) for the tree arrays. The expectation is that `((a,b)c)extra;` should become `(a,b)c` as `extra` only contains a single child. Interestingly, the collapse happens partially if there are multiple extras where `(((a,b)c)extra1)extra2;` becomes `((a,b)c)extra2;`.

```python
09:25:16 (daniel@sandbar):repicking> t = TreeNode.read(["((a,b)c)extra;"])

09:26:00 (daniel@sandbar):repicking> print(t.ascii_art())
                    /-a
-extra--- /c-------|
                    \-b

09:26:03 (daniel@sandbar):repicking> t.prune()

09:26:05 (daniel@sandbar):repicking> print(t.ascii_art())
                    /-a
-extra--- /c-------|
                    \-b
```

### PR DESCRIPTION
**Pull request by wasade**
Fixes #1247 and #807

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,8 @@
 * Fixed bug when using `Sequence.iter_kmers` on empty `Sequence` object. Previously this raised a `ValueError`, now it returns
 an empty generator.
 * Fixed minor bug where adding sequences to an empty `TabularMSA` with MSA-wide `positional_metadata` would result in a `TabularMSA` object in an inconsistent state. This could happen using `TabularMSA.append` or `TabularMSA.extend`. This bug only affects a `TabularMSA` object *without* sequences that has MSA-wide `positional_metadata` (for example, `TabularMSA([], positional_metadata={'column': []})`).
+* `TreeNode.distance` now handles the situation in which `self` or `other` are ancestors. ([#807](https://github.com/biocore/scikit-bio/issues/807))
```
Comment (jairideout) [inline CHANGELOG.md]: For both of these bugs, can you please describe what the previous behavior was? We're making an effort to note that for bug fixes in the changelog.

**Comment by jairideout** [comment]
Thanks for the fixes @wasade! I just had one changelog-related comment.

**Comment by wasade** [comment]
Should be good!

**Comment by jairideout** [comment]
Looks great, thanks @wasade!

**Pull request merged**